ui/src/app/tasks/task-definition/executions/task-definition-executions.component.ts

```diff
@@ -113,4 +213,62 @@ export class TaskDefinitionExecutionsComponent implements OnInit, OnDestroy {
     this.router.navigate([`tasks/executions/${taskExecution.executionId}`]);
   }
 
+
+  /**
+   * Update the list of selected checkbox
+   */
+  changeCheckboxes(taskExecutions) {
+
+    if (!taskExecutions) {
+      return;
+    }
+    const taskCheckable = taskExecutions.items;
+    if (taskCheckable.length !== this.form.checkboxes.length) {
+      return;
+    }
+    const value: Array<number> = taskCheckable
+      .map((ex, index) => {
+        if (this.form.checkboxes[index] && !ex.parentTaskExecutionId) {
+          return ex.executionId;
+        }
+      })
+      .filter((a) => a != null);
+    this.itemsSelected = value;
+  }
+
+  /**
+   * Starts the destroy process of multiple {@link TaskExecution}s
+   * by opening a confirmation modal dialog.
+   */
+  destroySelectedExecutions(page, params) {
+    const taskExecutions = page.items
+      .filter((item) => this.itemsSelected.indexOf(item.executionId) > -1);
+    this.destroyExecutions(taskExecutions, params);
+  }
+
+  /**
+   * Starts the destroy the {@link TaskExecution}s in parameter
```

Comment:
Maybe better: `Destroys the {@link TaskExecution}s that are provided as parameter. Will open a a confirmation modal dialog.`